src/coreclr/jit/eeinterface.cpp

```diff
@@ -226,21 +221,39 @@ void Compiler::eePrintMethod(StringPrinter*        printer,
                              CORINFO_CLASS_HANDLE  clsHnd,
                              CORINFO_METHOD_HANDLE methHnd,
                              CORINFO_SIG_INFO*     sig,
-                             bool                  includeNamespaces,
                              bool                  includeClassInstantiation,
                              bool                  includeMethodInstantiation,
                              bool                  includeSignature,
                              bool                  includeReturnType,
                              bool                  includeThisSpecifier)
 {
+    CorInfoHelpFunc helper = eeGetHelperNum(methHnd);
+    if (helper != CORINFO_HELP_UNDEF)
+    {
+        assert(helper < CORINFO_HELP_COUNT);
+        printer->Append(s_jitHelperNames[helper]);
+        return;
+    }
+
     if (clsHnd != NO_CLASS_HANDLE)
     {
-        eePrintType(printer, clsHnd, includeNamespaces, includeClassInstantiation);
+        eePrintType(printer, clsHnd, includeClassInstantiation);
         printer->Append(':');
     }
 
-    const char* methName = info.compCompHnd->getMethodName(methHnd, nullptr);
-    printer->Append(methName);
+    size_t requiredBufferSize;
+    char   buffer[256];
+    info.compCompHnd->printMethodName(methHnd, buffer, sizeof(buffer), &requiredBufferSize);
+    if (sizeof(buffer) <= requiredBufferSize)
+    {
+        printer->Append(buffer);
+    }
+    else
+    {
+        char* pBuffer = new (this, CMK_DebugOnly) char[requiredBufferSize];
+        info.compCompHnd->printMethodName(methHnd, pBuffer, requiredBufferSize);
+        printer->Append(pBuffer);
+    }
```

Comment:
This is a common pattern now. Might be nice to abstract it.

Reply:
Done, I added an `eeAppendPrint` function to do this. Also realized the `sizeof(buffer) <= requiredBufferSize` check is flipped.